src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
                           node_info_list,
                           name_table,
                           street_name_suffix_table,
-                          intersection_generator)
+                          intersection_generator),
+      geojson_lines{"sliproads-lines.geojson", node_info_list},
+      geojson_points{"sliproads-points.geojson", node_info_list}
 {
 }
 
-// included for interface reasons only
+// The intersection has to connect a Sliproad, see the example scenario below:
+// Intersection at `d`: Sliproad `bd` connecting `cd` and the road starting at `d`.
 bool SliproadHandler::canProcess(const NodeID /*nid*/,
                                  const EdgeID /*via_eid*/,
                                  const Intersection &intersection) const
 {
     return intersection.size() > 2;
 }
 
+// Detect sliproad b-d in the following example:
+//
+//       .
+//       e
+//       .
+//       .
+// a ... b .... c .
+//       `      .
+//         `    .
+//           `  .
+//              d
+//              .
+//
+// ^ a nid
+//    ^ ab source_edge_id
+//       ^ b intersection
 Intersection SliproadHandler::
-operator()(const NodeID, const EdgeID source_edge_id, Intersection intersection) const
+operator()(const NodeID /*nid*/, const EdgeID source_edge_id, Intersection intersection) const
 {
+    BOOST_ASSERT(intersection.size() > 2);
+
+    // Potential splitting / start of a Sliproad (b)
     auto intersection_node_id = node_based_graph.GetTarget(source_edge_id);
 
-    // if there is no turn, there is no sliproad
-    if (intersection.size() <= 2)
-        return intersection;
+    // Road index prefering non-sliproads (bc)
+    auto obvious = getObviousIndexWithSliproads(source_edge_id, intersection, intersection_node_id);
```

Comment:
since this returns an optional, I feel that `getObviousIndex` could be misleading as a name. We do not have a naming convention here, so far as I know. But how about `checkIfHasObviousIndexWithSliproads`?